routes/items.js

```diff
@@ -8,24 +8,66 @@ router.get("/", (req, res, next) => {
 });
 
 router.get("/:id", (req, res, next) => {
-  // TODO: complete writing this route handler
-  res.status(501).send({ error: 'route not yet implemented' });
+  const inputId = req.params.id;
+  const item = itemData.getById(inputId);
+  if (item === undefined) {
+    res.status(404).send(`no item with id ${inputId} found`);
+  }
+  else {
+    res.status(200).send(item);
+  }
 });
 
 router.post("/", (req, res, next) => {
-  itemData.create(req.body);
-  res.sendStatus(200);
+
+  if (!validateRequestBody(req.body)) {
+    res.status(400).send('invalid request body');
+  }
+  else {
+    itemData.create(req.body);
+    res.sendStatus(200);
+  }
 });
 
 router.put("/:id", (req, res, next) => {
-  // TODO: complete writing this route handler
-  res.status(501).send({ error: 'route not yet implemented' });
+  const inputId = req.params.id;
+
+  // Validating request body
+  if (!validateRequestBody(req.body)) {
+    res.status(400).send('invalid request body');
+  }
+  else {
+    itemData.updateById(inputId, req.body);
+    res.status(200).send();
```

Comment:
For cases where a status is all your returning, you can use the `sendStatus` method instead of chaining `status()` and `send()`